### PR TITLE
[DOCS] Clarify compatibility and relation between high level REST client and core version

### LINKED ISSUE #26142: HL Rest Client 6.0.0-beta1 is not full compatible with 5.5.1
Let's say you have a 5.5.1 cluster running. You want to run a QueryString query on it with:

```java
SearchResponse response = hlclient.search(
  new SearchRequest("foo").source(
    new SearchSourceBuilder().query(
      QueryBuilders.queryStringQuery("bar"))));
```

It fails because 6.0.0-beta1 QueryString object has now a `type` field which a 5.x version can not read:

```
ElasticsearchStatusException[Elasticsearch exception [type=parsing_exception, reason=[query_string] query does not support [type]]]
	at __randomizedtesting.SeedInfo.seed([1454848CA14180A5:1ED18A07CB044C51]:0)
	at org.elasticsearch.rest.BytesRestResponse.errorFromXContent(BytesRestResponse.java:177)
	at org.elasticsearch.client.RestHighLevelClient.parseEntity(RestHighLevelClient.java:522)
	at org.elasticsearch.client.RestHighLevelClient.parseResponseException(RestHighLevelClient.java:498)
	at org.elasticsearch.client.RestHighLevelClient.performRequest(RestHighLevelClient.java:405)
	at org.elasticsearch.client.RestHighLevelClient.performRequestAndParseEntity(RestHighLevelClient.java:378)
	at org.elasticsearch.client.RestHighLevelClient.search(RestHighLevelClient.java:319)
```

Ideally the JSon serialization of those objects should take care of this compatibility and be aware of the version currently running.

If we don't fix it, it will mean that the HL Rest client is somehow tight to the cluster version we are running.

### PR DESCRIPTION
**Pull request by javanna**
The client is forward compatible, while there may be bw comp issues even for non breaking changes made to ES core. That is why it's recommended to upgrade the client last and always have it in a version that's less or equal than the smallest node version.

Closes #26142

### CODE REVIEW

File: docs/java-rest/high-level/usage.asciidoc [hunk 1]
```diff
@@ -9,15 +9,25 @@ getting the artifact to using it in an application.
...
+new REST client.
+
+The High Level Client is guaranteed to be able to communicate with any Elasticsearch
+node running on the same major version and greater or equal minor version. It
+doesn't need to be in the same minor version as the Elasticsearch nodes it
+communicates with, as it is forward compatible meaning that it supports
+communicating with later versions of Elasticsearch than the one it was developed for.
+That means that the 6.0 client is able to communicate with any 6.x Elasticsearch node,
+while the 6.1 client is for sure able to communicate with 6.1, 6.2 and any later 6.x
+version, but there may be incompatibility issues when communicating with a previous
```
Comment (nik9000) [inline docs/java-rest/high-level/usage.asciidoc]: I think this should be two sentences or more sentences.

**Review by tlrx — APPROVED** [review]
LGTM, thanks, it's really better now.

**Comment by cbuescher** [comment]
I also think it is good but support Niks comment.

File: docs/java-rest/high-level/usage.asciidoc [hunk 2]
```diff
@@ -9,15 +9,25 @@ getting the artifact to using it in an application.
...
-
-The High Level Client is backwards compatible but can only communicate with Elasticsearch
-version 5.5 and onwards. The High Level Client is forward compatible as well, meaning that
-it supports communicating with a later version of Elasticsearch than the one it was developed
-for. It is recommended to upgrade the High Level Client when upgrading the Elasticsearch
-cluster to a new major version, as REST API breaking changes may cause unexpected results,
-and newly added APIs will only be supported by the newer version of the client. The client
-should be updated last, once all of the nodes in the cluster have been upgraded.
+and returns the same response objects. See the <<java-rest-high-level-migration>>
+if you are needing to migrate an application from `TransportClient` to the
```
Comment (cbuescher) [inline docs/java-rest/high-level/usage.asciidoc]: nit: I'm not a native speaker, so maybe @nik9000 can judge this better, but "you are needing to" sounds odd to me. For me "need to", "want to" or "are migrating" sounds better

File: docs/java-rest/high-level/usage.asciidoc (hunk 1 above)
Comment (javanna) [inline docs/java-rest/high-level/usage.asciidoc]: you mean paragraphs?

**Pull request merged**